### ISSUE TITLE
having trouble launching horizons 4.0 through min ED launcher (Epic games)

### ISSUE THREAD

**Issue by LXSTWXRDEN** [rfvgyhn/min-ed-launcher#59]
Hi. After installing horizons 4.0 and your launcher, I'm having trouble launching through it. 
I did everything according to the instructions for Epic and Windows.
The problem is that the game does not start through the minimum louncer. I will attach the log as well.
I would be grateful for your help in solving this problem.

Log min ed launcher:

2022-09-22 17:11:39.967 +03:00 [INF] Elite Runtime
    Platform: Epic
    CobraBay Version: 0.4.6725.0
    Products Dir: N:\EliteDangerous\Products
2022-09-22 17:11:40.814 +03:00 [DBG] Requesting epic token
2022-09-22 17:11:41.518 +03:00 [DBG] Requesting epic token success
2022-09-22 17:11:41.540 +03:00 [DBG] Authenticating via Epic
2022-09-22 17:11:42.457 +03:00 [DBG] Successfully authenticated
2022-09-22 17:11:42.463 +03:00 [INF] Logged in via Epic as: Oleg Truskov
2022-09-22 17:11:42.463 +03:00 [DBG] Getting authorized products
2022-09-22 17:11:43.668 +03:00 [DBG] Purchases Response:
{"purchases":[{"colour":"#f07b05","filter":"ed","directory":"elite-dangerous-64","serverargs":"","gameargs":"SeasonOne","sortkey":"06","product_name":"Elite Dangerous","product_sku":"FORC-FDEV-D-1010","template":"http://hosting.zaonce.net/launcher/elite/en.html"},{"colour":"#f07b05","filter":"eda","directory":"elite-dangerous-64","serverargs":"","gameargs":"CQCArena","sortkey":"20","product_name":"Elite Dangerous: Arena","product_sku":"FORC-FDEV-D-1012","template":"http://hosting.zaonce.net/launcher/elite-dangerous-arena/en.html"},{"colour":"#f07b05","filter":"ed","directory":"COMBAT_TUTORIAL_DEMO","serverargs":"","gameargs":"","sortkey":"30","product_name":"Single Player Combat Training","product_sku":"COMBAT_TUTORIAL_DEMO","template":"http://hosting.zaonce.net/launcher/elite/en.html"},{"colour":"#0a8bd6","filter":"edh","directory":"elite-dangerous-64","serverargs":"","gameargs":"","sortkey":"05","product_name":"Elite Dangerous: Horizons","product_sku":"FORC-FDEV-D-1013","template":"http://hosting.zaonce.net/launcher/elite-dangerous-horizons/en.html"},{"colour":"#f07b05","filter":"ed","directory":"elite-dangerous-odyssey-64","serverargs":"","gameargs":"SeasonTwo","sortkey":"04","product_name":"Elite Dangerous: Horizons (4.0)","product_sku":"FORC-FDEV-DO-38-IN-40","template":"http://hosting.zaonce.net/launcher/elite/en.html"}]}
2022-09-22 17:11:43.694 +03:00 [DBG] Authorized Products: Elite Dangerous: Horizons (4.0),Elite Dangerous: Horizons,Elite Dangerous,Elite Dangerous: Arena,Single Player Combat Training
2022-09-22 17:11:43.694 +03:00 [INF] Checking for updates
2022-09-22 17:11:43.702 +03:00 [DBG] Disabling 'Single Player Combat Training'. Unable to find product at 'N:\EliteDangerous\Products\COMBAT_TUTORIAL_DEMO\VersionInfo.txt'

**Comment by rfvgyhn** [comment]
Is that a full run of the log file? It seems to be missing the first line which should look like `[INF] Elite Dangerous: Minimal Launcher - [version]`. It also seems like there should be more lines at the end (unless it just crashes without writing more).

**Comment by LXSTWXRDEN** [comment]
Args: [|"/Epic"; "-AUTH_LOGIN=unused";
  "-AUTH_PASSWORD=d3d...f4d"; "-AUTH_TYPE=exchangecode";
  "-epicapp=9c2...593"; "-epicenv=Prod"; "-EpicPortal";
  "/autorun"; "/autoquit"; "/EDH"; "-epicusername=Weedmanov";
  "-epicuserid=792...8dd"; "-epiclocale=ru";
  "-epicsandboxid=3db...ac1"|]
    OS: Win64
    Env: 
    
2022-09-23 19:49:51.940 +03:00 [DBG] Reading settings from 'C:\Users\Олег\AppData\Local\min-ed-launcher\settings.json'
2022-09-23 19:49:52.520 +03:00 [DBG] Settings: Ok
  { Platform = Epic { ExchangeCode = "d3d...f4d"
                      Type = "exchangecode"
                      AppId = "9c2...593" }
    DisplayMode = Pancake
    AutoRun = true
    AutoQuit = true
    WatchForCrashes = false
    ProductWhitelist = seq ["edh"]
    ForceLocal = true
    CompatTool = None
    CbLauncherDir = "N:\EliteDangerous\"
    PreferredLanguage = None
    ApiUri = https://api.zaonce.net/
    Restart = None
    AutoUpdate = true
    MaxConcurrentDownloads = 4
    ForceUpdate = set []
    Processes = []
    FilterOverrides =
     seq [[FORC-FDEV-DO-1000, edo]; [FORC-FDEV-DO-38-IN-40, edh4]] }
2022-09-23 19:49:52.534 +03:00 [INF] Elite Runtime
    Platform: Epic
    CobraBay Version: 0.4.6725.0
    Products Dir: N:\EliteDangerous\Products
2022-09-23 19:49:53.524 +03:00 [DBG] Requesting epic token
2022-09-23 19:49:55.124 +03:00 [DBG] Requesting epic token success
2022-09-23 19:49:55.144 +03:00 [DBG] Authenticating via Epic
2022-09-23 19:49:56.068 +03:00 [DBG] Successfully authenticated
2022-09-23 19:49:56.073 +03:00 [INF] Logged in via Epic as: Oleg Truskov
2022-09-23 19:49:56.073 +03:00 [DBG] Getting authorized products
2022-09-23 19:49:57.327 +03:00 [DBG] Purchases Response:
{"purchases":[{"colour":"#f07b05","filter":"ed","directory":"elite-dangerous-64","serverargs":"","gameargs":"SeasonOne","sortkey":"06","product_name":"Elite Dangerous","product_sku":"FORC-FDEV-D-1010","template":"http://hosting.zaonce.net/launcher/elite/en.html"},{"colour":"#f07b05","filter":"eda","directory":"elite-dangerous-64","serverargs":"","gameargs":"CQCArena","sortkey":"20","product_name":"Elite Dangerous: Arena","product_sku":"FORC-FDEV-D-1012","template":"http://hosting.zaonce.net/launcher/elite-dangerous-arena/en.html"},{"colour":"#f07b05","filter":"ed","directory":"COMBAT_TUTORIAL_DEMO","serverargs":"","gameargs":"","sortkey":"30","product_name":"Single Player Combat Training","product_sku":"COMBAT_TUTORIAL_DEMO","template":"http://hosting.zaonce.net/launcher/elite/en.html"},{"colour":"#0a8bd6","filter":"edh","directory":"elite-dangerous-64","serverargs":"","gameargs":"","sortkey":"05","product_name":"Elite Dangerous: Horizons","product_sku":"FORC-FDEV-D-1013","template":"http://hosting.zaonce.net/launcher/elite-dangerous-horizons/en.html"},{"colour":"#f07b05","filter":"ed","directory":"elite-dangerous-odyssey-64","serverargs":"","gameargs":"SeasonTwo","sortkey":"04","product_name":"Elite Dangerous: Horizons (4.0)","product_sku":"FORC-FDEV-DO-38-IN-40","template":"http://hosting.zaonce.net/launcher/elite/en.html"}]}
2022-09-23 19:49:57.350 +03:00 [DBG] Authorized Products: Elite Dangerous: Horizons (4.0),Elite Dangerous: Horizons,Elite Dangerous,Elite Dangerous: Arena,Single Player Combat Training
2022-09-23 19:49:57.350 +03:00 [INF] Checking for updates
2022-09-23 19:49:57.353 +03:00 [DBG] Disabling 'Elite Dangerous: Horizons (4.0)'. Unable to find product at 'N:\EliteDangerous\Products\FORC-FDEV-DO-38-IN-40\VersionInfo.txt'
2022-09-23 19:49:57.360 +03:00 [DBG] Disabling 'Single Player Combat Training'. Unable to find product at 'N:\EliteDangerous\Products\COMBAT_TUTORIAL_DEMO\VersionInfo.txt'
2022-09-23 19:50:00.976 +03:00 [INF] Available Products:
	Elite Dangerous: Horizons FORC-FDEV-D-1013 Up to Date
	Elite Dangerous           FORC-FDEV-D-1010 Up to Date
	Elite Dangerous: Arena    FORC-FDEV-D-1012 Up to Date

**Comment by rfvgyhn** [comment]
It's hard to say. There are no WARN or ERR messages in the log file and it ends at different points in the launch process so I can't really tell why it's crashing.

If you remove the `/autoquit` flag, does it show any additional info compared to the log file?

**Comment by rfvgyhn** [comment]
Closing since no response. Feel free to re-open if you can provide more info.

**Comment by CmdrHope** [comment]
Ok, continuing  #66  here.

> Does the launcher just quit or does it just stop writing to the log file?

The Epic Games Launcher starts, then a terminal window of the min-ed-launcher shortly opens and closes again, then the Epic Games Launcher stays open and shows the launch button for Elite Dangerous. Clicking the launch button just opens and closes the terminal window again.

> Does it always fail in the same spot?

Yes, it does.

> Does it have the same issue if you remove the `/autorun` flag?

Yes, it does.

> Which version of wine are you using?

lutris-7.2-2-x86_64

> Which lutris installer did you use? I couldn't get any of the provided ones to work and had to cobble together my own. It's possible there's some wine environment difference that I don't have locally.

I used the `lutris_0.5.11_all.deb` they released on their github page.

> What game options do you have setup in lutris?

Pretty much the defaults but I disabled Fsync, FSR and GameMode.

> I added some additional logging. Can you try the [CI build](https://github.com/rfvgyhn/min-ed-launcher/actions/runs/3212048130) and post the new log?

Ok, here it is: [min-ed-launcher.log](https://github.com/rfvgyhn/min-ed-launcher/files/9742564/min-ed-launcher.log)
First I did a run `/autorun` and then another without.

**Comment by rfvgyhn** [comment]
> I used the lutris_0.5.11_all.deb they released on their github page

I wasn't clear on that. I meant which installer (yaml/json) for Elite did you use?

Another user looks to have a similar issue (#67) and I _think_ it's fixed with [this build](https://github.com/rfvgyhn/min-ed-launcher/actions/runs/3215453302). Can you try that?

Also, which version of windows is wine configured as? I think I ran into issues in the past when it was windows 10, but windows 7 worked.

**Comment by CmdrHope** [comment]
Ok, right after I have sent my last comment I tried a different wine version (`lutris-6.21-6-x86_64`) again, which definitely didn't work last week when I opened #66. But now it starts Elite Dangerous just fine. Then I tried [the newer build](https://github.com/rfvgyhn/min-ed-launcher/actions/runs/3215453302) which starts fine, too. Then I switched the wine version back to `lutris-7.2-2-x86_64` and starting Elite Dangerous fails again. However setting the wine configuration to windows 7 works on `lutris-7.2-2-x86_64` and then it starts the game again!

One little issue remains, after exiting Elite Dangerous the process stays open, so `/autoquit` doesn't seem to work properly. But that seems like a minor issue.

Finally I can run EDH4 on my Epic Account, too! Yeah! Thank you, great work!
